src/Discord/Internal/Gateway/Cache.hs

```diff
@@ -20,25 +20,19 @@ data Cache = Cache
      , cacheDMChannels :: !(M.Map ChannelId Channel)
      , cacheGuilds :: !(M.Map GuildId (Maybe (Guild, Maybe GuildCreateData)))
      , cacheChannels :: !(M.Map ChannelId Channel)
-     , cacheApplication :: !PartialApplication
+     , cacheApplication :: !FullApplication
      } deriving (Show)
 
 data CacheHandle = CacheHandle
   { cacheHandleEvents :: Chan (Either GatewayException EventInternalParse)
   , cacheHandleCache  :: MVar (Either (Cache, GatewayException) Cache)
   }
 
+createCache :: User -> FullApplication -> Cache
+createCache user app = Cache user M.empty M.empty M.empty app
+
 cacheLoop :: Bool -> CacheHandle -> Chan T.Text -> IO ()
-cacheLoop isEnabled cacheHandle log = do
-      ready <- readChan eventChan
-      case ready of
-        Right (InternalReady _ user _ _ _ _ pApp) -> do
-          putMVar cache (Right (Cache user M.empty M.empty M.empty pApp))
-          loop
-        Right r ->
-          writeChan log ("cache - stopping cache - expected Ready event, but got " <> T.pack (show r))
-        Left e ->
-          writeChan log ("cache - stopping cache - gateway exception " <> T.pack (show e))
+cacheLoop isEnabled cacheHandle _log = loop
```

Comment:
as my other comment said, a refactor could be that we do `when isEnabled loop`, which means that the cache thread won't just be spinning endlessly in the background. this would also simplify the `loop`.